### ISSUE TITLE
type mismatch false positive in array length

### ISSUE THREAD

**Issue by lovasoa** [intellij-rust/intellij-rust#4310]
## Environment

* **IntelliJ Rust plugin version:** 0.2.104.2132-192
* **Rust toolchain version:** 1.37.0 (eae3437df 2019-08-13) x86_64-apple-darwin
* **IDE name and version:** IC-192.6262.58
* **Operating system:** macOS 10.14.6

## Steps to reproduce
```rust
type T = [u8; 1 << 7u8];
```

![image](https://user-images.githubusercontent.com/552629/63710493-93362100-c839-11e9-94fd-196f16ed60b9.png)

**Comment by lovasoa** [comment]
I found the bug here: https://github.com/kaksmet/jpeg-decoder/blob/master/src/huffman.rs#L183-L184

**Comment by lovasoa** [comment]
This may be related:
<img width="408" alt="image" src="https://user-images.githubusercontent.com/552629/63724629-88d74f80-c858-11e9-83d0-0d8ad0090759.png">

**Comment by vlad20012** [comment]
Will be fixed with the new macro expansion engine. You can enable it as described in #3628
